test/Microsoft.AspNetCore.Localization.Tests/CookieRequestCultureProviderTest.cs

```diff
@@ -136,5 +145,62 @@ public class CookieRequestCultureProviderTest
                 var response = await client.GetAsync(string.Empty);
             }
         }
+
+        [Fact]
+        public async Task ParseInvalidCookieShouldLogWarning()
+        {
+            var sink = new TestSink(
+                TestSink.EnableWithTypeName<CookieRequestCultureProvider>,
+                TestSink.EnableWithTypeName<CookieRequestCultureProvider>);
+            var loggerFactory = new TestLoggerFactory(sink, enabled: true);
+            var builder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    var options = new RequestLocalizationOptions
+                    {
+                        DefaultRequestCulture = new RequestCulture("en-US"),
+                        SupportedCultures = new List<CultureInfo>
+                        {
+                            new CultureInfo("ar-YE")
+                        },
+                        SupportedUICultures = new List<CultureInfo>
+                        {
+                            new CultureInfo("ar-YE")
+                        }
+                    };
+                    var provider = new CookieRequestCultureProvider
+                    {
+                        CookieName = "Preferences"
+                    };
+                    options.RequestCultureProviders.Insert(0, provider);
+                    app.UseRequestLocalization(options);
+                    app.Run(context =>
+                    {
+                        var requestCultureFeature = context.Features.Get<IRequestCultureFeature>();
+                        var requestCulture = requestCultureFeature.RequestCulture;
+                        Assert.Equal("en-US", requestCulture.Culture.Name);
+                        return Task.FromResult(0);
+                    });
+                })
+                .ConfigureServices(services =>
+                {
+                    services.AddSingleton(typeof(ILoggerFactory), loggerFactory);
+                    services.AddSingleton(typeof(ILogger), loggerFactory.CreateLogger<CookieRequestCultureProvider>());
+                });
+
+            using (var server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                client.DefaultRequestHeaders.Add("Cookie", new CookieHeaderValue("Preferences", "c=??|uic=en-US").ToString());
```

Comment:
The test name definitely helps, but it isn't obvious at first where in the code the "invalid cookie" is - or why it's invalid.